### PR TITLE
Add support for global allowed IP list

### PR DESCRIPTION
**Pull request by tomaisthorpe**
## Summary by Aikido
|  Security Issues: 0 | 🔍 Quality Issues: 1 |  Resolved Issues: 0 |
| :--- | :--- | :--- |

**🚀 New Features**
* Added global allowed IP list and enforcement checks across requests

**⚡ Enhancements**
* Implemented private and local IP detection and always-allowed behavior
* Integrated allowed IP checks into request initialization and responses

**🔧 Refactors**
* Refactored cloud list types to consolidate IP list structures

<sup>[More info](https://app.aikido.dev/featurebranch/scan/72182268?groupId=6)</sup>

### CODE REVIEW

File: internal/http/on_init_request.go [hunk 1]
```diff
@@ -24,8 +24,16 @@ func OnInitRequest(ctx context.Context) *Response {
 		return nil
 	}
 
-	// Blocked IP lists (e.g. known threat actors, geo blocking, ...)
 	ip := reqCtx.GetIP()
+
+	// Allowed IP list, global list for allowing traffic by country
+	if ipAllowed := config.IsIPAllowed(ip); !ipAllowed {
+		msg := fmt.Sprintf("Your IP address is not allowed to access this resource. (Your IP: %s)", ip)
```
Comment (bitterpanda63) [inline internal/http/on_init_request.go]:
```suggestion
		msg := fmt.Sprintf("Your IP address is not allowed. (Your IP: %s)", ip)
```

Comment (bitterpanda63) [reply internal/http/on_init_request.go]: the "access this resource" is specifically for admin portals (so allowed ip lists that are endpoint-level)

File: internal/agent/ipaddr/private_ip_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package ipaddr
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsPrivateIP(t *testing.T) {
+	t.Run("identifies private IPv4 addresses", func(t *testing.T) {
```
Comment (bitterpanda63) [inline internal/agent/ipaddr/private_ip_test.go:10]: Let's add test cases for IPv6-mapped IPv4 ip addresses?

File: internal/agent/config/service_config.go [hunk 3]
```diff
@@ -134,6 +152,36 @@ func IsIPBlocked(ip string) (bool, string) {
...
+	if len(serviceConfig.AllowedIPs) == 0 {
+		return true
+	}
+
+	// Always allow private/local IP addresses (matches Node.js behavior)
+	if ipaddr.IsPrivateIP(ip) {
+		return true
+	}
+
+	ipAddress, err := ipaddr.Parse(ip)
```
Comment (bitterpanda63) [inline internal/agent/config/service_config.go:170]: let's make sure this doesnt block weird edge case IP addresses (also test ipv6-mapped ipv4 addresses)

File: internal/http/on_init_request.go (hunk 1 above)
Comment (tomaisthorpe) [reply internal/http/on_init_request.go]: Updated, looks like it needs to be updated in `firewall-node` too.

File: internal/agent/ipaddr/private_ip_test.go (hunk 2 above)
Comment (tomaisthorpe) [reply internal/agent/ipaddr/private_ip_test.go:10]: added!

**Review by bitterpanda63 — APPROVED** [review]
